### PR TITLE
Restore survival tests into test folder

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Move survival tests from `tests-legacy` into `tests`, provide them a new location, documentation and dedicated setup file
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis should be green

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11853)

### CODE REVIEW

**Comment by matks** [comment]
Note: I need to update this PR to make Travis run these

**Comment by matks** [comment]
Travis "unit" build will now run SurvivalTests. Waiting for travis feedback ...

**Comment by matks** [comment]
Survival tests were run successfully ✅but I failed the lint steps 😁

File: tests/check_integration.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+composer run-script integration-survival-tests --timeout=0;
+UNIT=$?
+
+if [[ "$UNIT" == "0" ]]; then
+  echo -e "\e[INTEGRATION TESTS OK"
```
Comment (PierreRambaud) [inline tests/check_integration.sh:7]: You forget to close the color style. And forget to add a color when it is ok.

Comment (PierreRambaud) [inline tests/check_integration.sh:7]:
```suggestion
  echo -e "\e[92mINTEGRATION TESTS OK\e[0m"
```

File: tests/check_integration.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+else
+  echo -e "\e[91mINTEGRATION TESTS FAILED"
```
Comment (PierreRambaud) [inline tests/check_integration.sh:9]:
```suggestion
  echo -e "\e[91mINTEGRATION TESTS FAILED\e[0m"
```

**Comment by matks** [comment]
@PierreRambaud I applied your suggestions 👍

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Hey, a little rebase please? :)

**Comment by PierreRambaud** [comment]
Still needed @matks ?

**Comment by matks** [comment]
I cant find time to continue this PR.

**Pull request closed without merging**